### PR TITLE
Work squalene automatic mapping mission creation

### PR DESCRIPTION
**Pull request by Squalene**
Enables the creation of a mapping mission from a parallelogram.

Caveat: all computations are done in meters. An adapter/provider must be built to convert LatLng to meters.

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/Parallelogram.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import ch.epfl.sdp.drone3d.mission.Point.Companion.distance
+import com.mapbox.mapboxsdk.geometry.LatLng
+
+/**
+ * Utility class to represent a parallelogram
+ */
+data class Parallelogram(val origin:Point, val dir1Span:Vector, val dir2Span:Vector){
+
+    constructor(origin: Point,dir1Point:Point,dir2Point:Point): this(origin,Vector(origin,dir1Point),Vector(origin,dir2Point))
```
Comment (GabrielFleischer) [inline app/src/main/java/ch/epfl/sdp/drone3d/mission/Parallelogram.kt:16]: Maybe rename dir1Point and dir2Point to adjacentPoint1 and adjacentPoint2

File: app/src/main/java/ch/epfl/sdp/drone3d/service/storage/data/MappingMission.kt [hunk 2]
```diff
@@ -4,7 +4,6 @@
  */
 
 package ch.epfl.sdp.drone3d.service.storage.data
-
```
Comment (GabrielFleischer) [inline app/src/main/java/ch/epfl/sdp/drone3d/service/storage/data/MappingMission.kt:7]: A space between package and import should be kept

**Review by GabrielFleischer — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/Parallelogram.kt (hunk 1 above)
Comment (Squalene) [reply app/src/main/java/ch/epfl/sdp/drone3d/mission/Parallelogram.kt]: Good point, done

File: app/src/main/java/ch/epfl/sdp/drone3d/service/storage/data/MappingMission.kt (hunk 2 above)
Comment (Squalene) [reply app/src/main/java/ch/epfl/sdp/drone3d/service/storage/data/MappingMission.kt]: Done

**Pull request merged**